pkg/tasks/tasks.go

```diff
@@ -104,7 +104,14 @@ func WithHostnameOSAndProbes(t Tasks) Tasks {
 // WithFullInstall with install binaries (using WithBinariesOnly) and
 // orchestrate complete cluster init
 func WithFullInstall(t Tasks) Tasks {
-	return WithBinariesOnly(t).
+	return Tasks{
+		{
+			Fn: func(s *state.State) error {
+				return s.RunTaskOnAllNodes(disableNMCloudSetup, state.RunParallel)
+			},
+			ErrMsg: "failed to disable nm-cloud-setup",
+		},
+	}.append(WithBinariesOnly(t)...).
```

Comment:
sooo... this would work.

Inside disableNMCloudSetup you are checking the OS of the node and if it was initialized (in liveCluster), but non of those tasks has been run at this point since `WithBinariesOnly()` has inside it `WithHostnameOSAndProbes`, so basically WithHostnameOSAndProbes is a prerequisite to `disableNMCloudSetup`

Reply:
The result will be, the RHEL specific script from `disableNMCloudSetup` will never run.

Reply:
Regarding the `IsInitialized` that's ok, since the value is false and that's what I want. if the node is not initiaized then run the script. about the os, do you really need a task to know what OS is running ? don't you have this information already from tf?

Reply:
@kron4eg is right. You check the operating system in `pkg/tasks/prerequisites.go` on L115, so your function is not going to execute if you don't determine the OS first. We have two options:

1) Run `WithBinaries` before running `disableNMCloudSetup`
2) Replace `WithBinaries` in a way that we first run `WithHostnameOSAndProbes`, then `disableNMCloudSetup`, and at the end `installPrerequisites`

If I'm not mistaken, you mentioned once that the first approach might not always work correctly, so if that's the case, the second approach would be better.